### PR TITLE
LUCENE-10131: Add backcompat indices for 8.10 and add LUCENE_8_10_0 to Version

### PR DESCRIPTION
**Pull request by thelabdude**
Manually fixing up some steps missed during the 8.10 release process.

I don't think this is correct because 8.10 is not listed in the `oldNames` array in `TestBackwardsCompatibility`. Somebody who understands how the `testAllVersionsTested` method is supposed to work on main (see comment about TRUNK in the code) needs to review.

### CODE REVIEW

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java [hunk 1]
```diff
@@ -383,6 +383,7 @@ public void testCreateEmptyIndex() throws Exception {
     "sorted.8.0.0",
     "sorted.8.1.0",
     "sorted.8.1.1",
+    "sorted.8.10.0",
```
Comment (madrob) [inline lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java:386]: I think you also need an addition above to `oldNames`

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java [hunk 2]
```diff
@@ -685,7 +686,7 @@ public void testAllVersionHaveCfsAndNocfs() {
   }
 
   public void testAllVersionsTested() throws Exception {
-    Pattern constantPattern = Pattern.compile("LUCENE_(\\d+)_(\\d+)_(\\d+)(_ALPHA|_BETA)?");
+    Pattern constantPattern = Pattern.compile("LUCENE_(\\d+)_(\\d{1,2})_(\\d+)(_ALPHA|_BETA)?");
```
Comment (madrob) [inline lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java:689]: I don't think this is necessary.

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java (hunk 1 above)
Comment (thelabdude) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java:386]: yes, that's the problem @madrob , see my comment in the PR description!

File: lucene/core/src/java/org/apache/lucene/util/Version.java [hunk 3]
```diff
@@ -168,6 +168,13 @@
    */
   @Deprecated public static final Version LUCENE_8_9_0 = new Version(8, 9, 0);
 
+  /**
+   * Match settings and bugs in Lucene's 8.10.0 release.
+   *
+   * @deprecated (8.11.0) Use latest
+   */
+  @Deprecated public static final Version LUCENE_8_10_0 = new Version(8, 10, 0);
```
Comment (madrob) [inline lucene/core/src/java/org/apache/lucene/util/Version.java:176]: I think you need to add an 8_11_0 version. A long time ago, when this all worked properly (like 8.2, I went spelunking in history) we would add the next minor immediately after a release.

Comment (janhoy) [reply lucene/core/src/java/org/apache/lucene/util/Version.java:176]: In main branch this looks good, with LUCENE_8_10_0 being added only. In branch_8x there would be a LUCENE_8_11_0, but that is [already OK](https://github.com/apache/lucene-solr/commit/08cdb36661957475359f815dda1e38915a6e331d)

Comment (thelabdude) [reply lucene/core/src/java/org/apache/lucene/util/Version.java:176]: It seems 8_11 is needed for main too, I don't know why after 8.9, it worked without an 8_10 but I'm already frustrated by this split repo and the release process being broken

Comment (thelabdude) [reply lucene/core/src/java/org/apache/lucene/util/Version.java:176]: I added this JIRA to track improving the release manager so this doesn't happen again: https://issues.apache.org/jira/browse/LUCENE-10141

**Pull request merged**